charts/ai-software-templates/chatbot/templates/application-gitops-job.yaml

```diff
@@ -12,7 +12,85 @@ spec:
+              export TEKTON_FILE_APP_NAMESPACE_REPLACEMENT="application_namespace_replace"
+              export TEKTON_FILE_QUAY_ACCOUNT_REPLACEMENT="quay_account_replace"
+              echo APP_NAME is "$APP_NAME"
+              echo APP_NAMESPACE is "$APP_NAMESPACE"
+              echo GITHUB_ORG_NAME is "$GITHUB_ORG_NAME"
+              echo GITHUB_SOURCE_REPO is "$GITHUB_SOURCE_REPO"
+              echo GITHUB_TETKON_SOURCE_REPO is "$GITHUB_TETKON_SOURCE_REPO"
+              echo GITHUB_DEFAULT_BRANCH is "$GITHUB_DEFAULT_BRANCH"
+              echo QUAY_ACCOUNT_NAME is "$QUAY_ACCOUNT_NAME"
+
+              export GH_TOKEN=$GITHUB_TOKEN
+              export GH_REPO="github.com/$GITHUB_ORG_NAME/$APP_NAME"
+
+              export HOME=/tmp
+
+              gh auth login -p https
+              gh auth setup-git --hostname=github.com
+
+              gh repo view "$GH_REPO" --json id 2> /dev/null
+
+              return_code=$?
+
+              if [ $return_code == 0 ]; then
+                echo Repo "$GH_REPO" already exists
+                exit 0
+              else
+                gh repo create "$GH_REPO" --public --gitignore Python --license Apache-2.0 || exit
+                gh repo view "$GH_REPO" --json id 2> /dev/null
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo Repo create of "$GH_REPO" could not be confirmed
+                  exit $return_code
+                fi
+              fi
+
+              gh repo clone "$GITHUB_SOURCE_REPO" app-source -- --no-tags --branch="$GITHUB_DEFAULT_BRANCH" --single-branch
+              gh repo clone "$GITHUB_TETKON_SOURCE_REPO" tekton-source -- --no-tags --single-branch
+              gh repo clone "$GH_REPO" "$APP_NAME" -- --no-tags --branch="$GITHUB_DEFAULT_BRANCH" --single-branch
+
+              mkdir -p "$APP_NAME"/.tekton
+
+              cd tekton-source/pac/pipelineRuns/.tekton || exit
+              sed -i 's@'"$TEKTON_FILE_APP_NAME_REPLACEMENT"'@'"$APP_NAME"'@g' docker-push.yaml
+              sed -i 's@'"$TEKTON_FILE_APP_NAMESPACE_REPLACEMENT"'@'"$APP_NAMESPACE"'@g' docker-push.yaml
+              sed -i 's@'"$TEKTON_FILE_QUAY_ACCOUNT_REPLACEMENT"'@'"$QUAY_ACCOUNT_NAME"'@g' docker-push.yaml
+              cd - || exit
+
+              cd app-source/chatbot || exit
+              tar cf - ./* | (cd ../../"$APP_NAME" || exit; tar xf -)
+              cd ../../tekton-source/pac/pipelineRuns/.tekton || exit
+              tar cf - ./* | (cd ../../../../"$APP_NAME"/.tekton || exit; tar xf -)
+              cd ../../pipelines || exit
+              tar cf - ./* | (cd ../../../"$APP_NAME"/.tekton || exit; tar xf -)
+              cd ../tasks || exit
+              tar cf - ./* | (cd ../../../"$APP_NAME"/.tekton || exit; tar xf -)
+              cd ../../../"$APP_NAME" || exit
+
+              git config user.email  "{{ .Values.gitops.gitAuthorEmail }}"
+              git config user.name  "{{ .Values.gitops.gitAuthorName }}"
```

Comment:
Do we need to provide these values? We are using the Git Application, I wonder if we can just use the account associated with that?

For instance, this is the committer using the py script with the existing helm chart https://github.com/redhat-appstudio-mjf/helm-chart-chatbot/commits/main/

Reply:
I got a complaint from `git` it it was not set.  Though perhaps that has something to do with how I set up my Github application?

When I hit ^^ I speculated that the git python lib used in developer-images sets it under the covers.

Reply:
Any chance @maysunfaisal you could check out my PR into a local branch and try it with your github app etc. but with those 2 git config calls commented out?

If not, maybe @thepetk could try it when he gets back form PTO

That would narrow down if it is a 
- python git lib vs. git cmd thing
- or how I set up my github app

I'll poke around my github app / token / etc. again tomorrow see if I missed how to set that stuff

Reply:
Also, I may have figured out what might be the analogous `git commit` option that perhaps corresponds to that python code 

```
_gh_new_commit = target_repo.create_git_commit(
            "Copy repo content", _gh_new_tree, [_gh_parent_commit]
        )
```

it may be , from the man page / help for `git commit`

```
       -C <commit>, --reuse-message=<commit>
           Take an existing commit object, and reuse the log message and the authorship information (including the timestamp) when creating the commit.
```

Maybe the parent commit from the python lines up with the commit you "reuse".  

I'll experiment with that as well as revisit my github application, etc.

Reply:
@gabemontero I'm currently seting up everything to test the updated helm chart as part of my review. I'll get back to you with detailed results with/without setting up the author and email. FWIW I was also expecting that those lines would be required in this conversion. It's also handled similarly in https://github.com/redhat-ai-dev/rhdh-pipelines/blob/main/pac/tasks/update-deployment.yaml

Reply:
I got it sorted out @maysunfaisal  .... part of next push